src/blockedtuple.jl

```diff
@@ -0,0 +1,95 @@
+# This file defines BlockedTuple, a Tuple of heterogeneous Tuple with a BlockArrays.jl
+# like interface
+
+using BlockArrays: Block, BlockArrays, BlockIndexRange, BlockRange, blockedrange
+
+#
+# ==================================  AbstractBlockTuple  ==================================
+#
+abstract type AbstractBlockTuple end
+
+# Base interface
+Base.axes(bt::AbstractBlockTuple) = (blockedrange([blocklengths(bt)...]),)
+
+Base.copy(bt::AbstractBlockTuple) = copy.(bt)
```

Comment:
So interestingly, Julia doesn't define `copy(::Tuple)`:
```julia
julia> copy((1, 2, 3))
ERROR: MethodError: no method matching copy(::Tuple{Int64, Int64, Int64})
[...]
```
probably because `copy` is shallow anyway, so maybe we should go the same route and not define `copy(::AbstractBlockTuple)`.